hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -134,12 +161,199 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
         LOG.info("No Instants to archive");
       }
 
+      if (config.getArchiveAutoMergeEnable()) {
+        mergeArchiveFilesIfNecessary(context);
+      }
       return success;
     } finally {
       close();
     }
   }
 
+  private void mergeArchiveFilesIfNecessary(HoodieEngineContext context) throws IOException {
+    Path planPath = new Path(metaClient.getArchivePath(), mergeArchivePlanName);
+    // Flush reminded content if existed and open a new write
+    reOpenWriter();
+    // List all archive files
+    FileStatus[] fsStatuses = metaClient.getFs().globStatus(
```

Comment:
do you think we should order these files in some order. like file sizes or mod time. Just incase there are too many files for the first time when this config is enabled, it might take few merge archives to merge all small archive files. I could not think of any correctness issue. but wanted to remind just incase.

Reply:
Nice catch here, I just find out that maybe it's important to keep the original instants order of small archive files.
https://github.com/apache/hudi/blob/b5f05fd153df29a8be377404a14a0ced2f00b4bf/hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieArchivedTimeline.java#L219
When load archived instants, hoodie will use this order to optimize skipping reading unnecessary archived files
https://github.com/apache/hudi/blob/b5f05fd153df29a8be377404a14a0ced2f00b4bf/hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieArchivedTimeline.java#L243

So just use the same order compactor here.
What do you think? :)

Reply:
yes, sounds good to me.